packages/phone-number-privacy/src/database/wrappers/account.ts

```diff
@@ -37,11 +53,36 @@ async function incrementExistingQueryCount(account: string) {
 }
 
 async function addNewQueryCount(account: string) {
-  const data = {
-    [ACCOUNTS_COLUMNS.address]: account,
-    [ACCOUNTS_COLUMNS.createdAt]: new Date(),
-    [ACCOUNTS_COLUMNS.numLookups]: 1,
+  const data = new Account(account)
+  data[ACCOUNTS_COLUMNS.numLookups] = 1
+  return updateRecord(data)
+}
+
+/*
+ * Set did matchmaking to true in database.  If record doesn't exist, create one.
+ */
+export async function setDidMatchmaking(account: string) {
+  await ((await setExistingDidMatchmaking(account)) || (await addNewRecordWithMatchmaking(account)))
```

Comment:
what's the intention here? Is it to try setting it and then add record if it fails?

Reply:
correct -- from my google searching, I could not find another way to do update-if-exists + insert-if-does-not-exist.

if you have thoughts on a better way, let me know!

Reply:
I would recommend just querying first rather than relying on update to return falsy values to get this or to work. 
Something like:
`getDatabase().select('address').from<Account>(ACCOUNTS_TABLE).where(ACCOUNTS_COLUMNS.address, account)`
then if that returns something, update it, otherwise create it. 

Semi-related, just as a point of style using multiple awaits on one line is possible but not great for readability and debugability :)

Reply:
sure --
i guess the benefit of the current implementation is that if it updates, then it only ever makes one db request.  if it queries first -- then there are always 2 request.

but by querying first, it is more clear what is happening, and that other update errors wouldn't cause subsequent insert request